state/useProjectFiltersState.ts

```diff
@@ -5,9 +5,24 @@ import { create } from "zustand"
 import { ProjectInterface } from "@/lib/types"
 import { uniq } from "@/lib/utils"
 
+export type ProjectSortBy = "random" | "asc" | "desc" | "relevancy"
 export type ProjectFilter = "keywords" | "builtWith" | "themes"
 export type FiltersProps = Record<ProjectFilter, string[]>
 
+interface ProjectInterfaceScore extends ProjectInterface {
+  score: number
+}
+
+export const SortByFnMapping: Record<
+  ProjectSortBy,
+  (a: ProjectInterfaceScore, b: ProjectInterfaceScore) => number
+> = {
+  random: () => Math.random() - 0.5,
+  asc: (a, b) => a.name.localeCompare(b.name),
+  desc: (a, b) => b.name.localeCompare(a.name),
+  relevancy: (a, b) => b?.score - a?.score, // sort from most relevant to least relevant
```

Comment:
```suggestion
  relevance: (a, b) => b?.score - a?.score, // sort from most relevant to least relevant
```

also how are the project scores determined

Reply:
@thebeyondr 

The score is determined by `fuse.js` the library that is used for searching matches.

> includeScore: Whether the score should be included in the result set. A score of 0 indicates a perfect match, while a score of 1 indicates a complete mismatch.

ref: https://www.fusejs.io/api/options.html#includescore  

Also 
- This PR this `includeScore` is enabled: https://github.com/privacy-scaling-explorations/website-v2/pull/81/files#diff-a085d3bde4c211cb48b341552b0348bb197ea68add00832179f87c2a718c10b0R161 

- `score` is part of `project` properties: https://github.com/privacy-scaling-explorations/website-v2/pull/81/files#diff-a085d3bde4c211cb48b341552b0348bb197ea68add00832179f87c2a718c10b0R168

Reply:
@kalidiagne Yeah ik that just making sure "Relevance" is only available within the context of search/filter